src/components/Confirm/index.js

```diff
@@ -225,8 +244,21 @@ class Confirm extends Component  {
         const data = { challengeData, blob, language, sdkMetadata}
         uploadLiveVideo(data, token, this.onApiSuccess, this.onApiError)
       } else {
-        const data = { file: blob, sdkMetadata}
-        uploadLivePhoto(data, token, this.onApiSuccess, this.onApiError)
+        Promise.all([
+          uploadSnapshotIfPresent({
```

Comment:
Be careful here as these will be running in parallel. If you want to guarantee without a doubt that that the snapshot request completes first, then you'll want to chain the promises. It'll be a lot slower, though :(

Reply:
Ohh, my bad, I thought this was synchronous. Yes, for this first iteration we need to make sure that the snapshot requests always completes first. I will chain them then.